### DISCUSSION TITLE
Dynamic provider configs

### DISCUSSION THREAD

**Discussion by binaryfire** [prism-php/prism#89] [Ideas]
Great package TJ!

Would it be possible to add a way to use dynamic provider configs without having to override config values? Config overrides are really problematic with Octane.

For example, this is how the Socialite Providers package does it:  https://socialiteproviders.com/usage/#overriding-a-config

This would make it easier to use the package in a multi-tenant context, where each tenant brings their own keys.

**Comment by sixlive** [comment]
I think this is something we could accomplish! I think the use case makes sense.

**Comment by sixlive** [comment]
Shipped in [v0.21.0](https://github.com/echolabsdev/prism/releases/tag/v0.21.0)

```php
Prism::text()
    ->usingProviderConfig(['api_key' => '1234']);

Prism::structured()
    ->usingProviderConfig(['api_key' => '1234']);
```
